### PR TITLE
mc/change typescript

### PR DESCRIPTION
**Pull request by Klochette**

### CODE REVIEW

File: src/components/extensionEditor/attributeEditor/AttributeEditor.tsx [hunk 1]
```diff
@@ -45,52 +39,51 @@ const AttributeEditor: React.FC<AttributeEditorProps> = ({
   let extensionElement = getElementById("Extension", "differential");
```
Comment (simonvadee) [inline src/components/extensionEditor/attributeEditor/AttributeEditor.tsx:39]: doesn't `getElementById` always return the same type (like `ElementDefinition` or something) ? why isn't the function return type specified ?
If extensionElement is statically typed, you wouldn't have to specify "as string" or "as number" each time you mention it.

File: src/components/structuredefSettings/StructuredefSettings.tsx [hunk 2]
```diff
@@ -53,33 +47,24 @@ const StructuredefSettings: React.FC<StructuredefSettingsProps> = ({
       e.preventDefault();
       const structureDefinitonToEdit = { ...structureDefinition };
       // content required
-      structureDefinitonToEdit.url = url as Uri.AsObject;
-      structureDefinitonToEdit.name = name as String.AsObject;
-      structureDefinitonToEdit.kind = kind as StructureDefinition.KindCode.AsObject;
-      structureDefinitonToEdit.abstract = abstract as Boolean.AsObject;
-      structureDefinitonToEdit.status = status as StructureDefinition.StatusCode.AsObject;
+      structureDefinitonToEdit.url = url as string;
```
Comment (simonvadee) [inline src/components/structuredefSettings/StructuredefSettings.tsx:50]: why do you need to "cast" the type ? shouldn't typescript be aware that `url` is a string ?

**Review by nriss — APPROVED** [review]
It looks so better than before ! Good job !
I think there are some useless casts when variable are already strings

File: src/components/extensionEditor/attributeEditor/AttributeEditor.tsx (hunk 1 above)
Comment (nriss) [reply src/components/extensionEditor/attributeEditor/AttributeEditor.tsx:39]: Right, it always return an ElementDefinition fhir type.
I think we can type the output of the func with IElementDefinition

**Pull request merged**